core/src/commonMain/kotlin/com/powersync/sync/SyncOptions.kt

```diff
@@ -2,6 +2,37 @@ package com.powersync.sync
 
 import com.powersync.ExperimentalPowerSyncAPI
 import com.powersync.PowerSyncDatabase
+import io.ktor.client.HttpClient
+import io.ktor.client.HttpClientConfig
+import kotlin.experimental.ExperimentalObjCRefinement
+import kotlin.native.HiddenFromObjC
+
+/**
+ * Configuration options for the [PowerSyncDatabase.connect] method, allowing customization of
+ * the HTTP client used to connect to the PowerSync service.
+ */
+@OptIn(ExperimentalObjCRefinement::class)
+public sealed class SyncClientConfiguration {
+    /**
+     * Extends the default Ktor [HttpClient] configuration with the provided block.
+     */
+    @HiddenFromObjC
+    public class ExtendedConfig(
+        public val block: HttpClientConfig<*>.() -> Unit,
+    ) : SyncClientConfiguration()
+
+    /**
+     * Provides an existing [HttpClient] instance to use for connecting to the PowerSync service.
+     * This client should be configured with the necessary plugins and settings to function correctly.
+     * The HTTP client requirements are delicate and subject to change throughout the SDK's development.
+     * The [configureSyncHttpClient] function can be used to configure the client for PowerSync.
```

Comment:
Perhaps also mention explicitly that the SDK will not call that function or apply any further configuration to the provided client.

Reply:
From a quick check, it seems that it's not possible to edit an existing client, but I might have missed something. I've updated the docs with an explicit mention.